server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -290,7 +291,7 @@ private void executeRequest(Task task,
                 if (shouldMinimizeRoundtrips(rewritten)) {
                     final TaskId parentTaskId = task.taskInfo(clusterService.localNode().getId(), false).getTaskId();
                     ccsRemoteReduce(parentTaskId, rewritten, localIndices, remoteClusterIndices, timeProvider,
-                        searchService.aggReduceContextBuilder(rewritten),
+                        searchService.aggReduceContextBuilder(((CancellableTask) task)::isCancelled, rewritten),
```

Comment:
This line is not clear me, can we assume that all Tasks here are instances of CancellableTask? maybe we should add a comment regarding the assumption.

Reply:
It can be only the task produced by SearchRequest which is SearchTask which is in turn CancellableTask. We can probably change it in the signature to make it clearer.

Reply:
+1 to change the signature if the provided Task is always a SearchTask.

Reply:
Do you need this cast any more now that you've changed the signature?